src/wallet/index.ts

```diff
@@ -48,33 +57,39 @@ export class Wallet {
   ensureUserLoggedIn = async (
     loginProvider: Web3AuthLoginProvider,
   ): Promise<AccountInfo> => {
-    if (
-      this._accountInfo &&
-      this._accountInfo.web3authUserInfo.loginProvider === loginProvider
-    ) {
-      return this._accountInfo
-    }
-    if (await this.web3AuthConnection.isLoggedIn()) {
-      // re-login in web3Auth with the previously logged user to get user info
-      const userInfo = await this.web3AuthConnection.reLogin()
-      // login only if the login provider of the previous user match the current login provider
-      if (userInfo.userMetadata.loginProvider === loginProvider) {
-        return this.login(userInfo)
+    try {
+      if (
+        this._accountInfo &&
+        this._accountInfo.web3authUserInfo.loginProvider === loginProvider
+      ) {
+        return this._accountInfo
+      }
+      if (await this.web3AuthConnection.isLoggedIn()) {
+        // re-login in web3Auth with the previously logged user to get user info
+        const userInfo = await this.web3AuthConnection.reLogin()
+        // login only if the login provider of the previous user match the current login provider
+        if (userInfo.userMetadata.loginProvider === loginProvider) {
+          return this.login(userInfo)
+        }
       }
+      // if not logged in, do fresh login
+      const userInfo = await this.web3AuthConnection.login(loginProvider)
+      return this.login(userInfo)
+    } catch (e) {
+      return this._callbacks.onLoginFail(e)
     }
-    // if not logged in, do fresh login
-    const userInfo = await this.web3AuthConnection.login(loginProvider)
-    return this.login(userInfo)
   }
 
   private login = async (userInfo: {
```

Comment:
I'd expect here also some `getCurrentUserInfo()` method to retrieve data about the user like which provider did they use to login (google, fb, ...), what's their email, etc - basically just proxy the web3auth login info to the app so that the user of the library has a way to expose it in the UI